R2Northstar/NorthstarMods
# Allow executing ``npc_pilot_elite`` NPCs

Allows players to execute pilot NPCs. Since every mod that uses ``npc_pilot_elite`` NPCs for something (see Gruntmode) , also adds similar changes to this, i think it'd be nice to bring this change into Northstar officially.

Before:
https://github.com/user-attachments/assets/489cc6f1-bc9f-4254-88a5-c30c7e30f6d6

After:
https://github.com/user-attachments/assets/b1263242-c6e8-403d-941c-8bed822ef5cc

Also added missing gib + headshot fx, to prevent weirdness.

Before:
https://github.com/user-attachments/assets/d8874ab2-fdeb-48eb-9db8-10526770f717

After:
https://github.com/user-attachments/assets/33eca4fd-3ec4-49a7-805a-024093efd068

HOW TO TEST:
1. Spawn a pilot NPC (``ent_create npc_pilot_elite``)
2. Try to execute them. Without this pr, you'll just punch them. With this pr, you'll properly execute them as expected.

Know issues:

* If firstperson executions are enabled, the game will create a duplicate pilot ragdoll after the animation's finished. See https://github.com/R2Northstar/NorthstarMods/issues/885
https://github.com/user-attachments/assets/68375bb5-29ed-4f59-b5fb-a5eb0a0fcc44
